app/scripts/metamask-controller.js

```diff
@@ -254,6 +266,22 @@ module.exports = class MetamaskController {
     this.configManager.useEtherscanProvider()
     extension.runtime.reload()
   }
+
+  buyEth (address, amount) {
+    if (!amount) amount = '5'
+
+    var network = this.idStore._currentState.network
+    var url = `https://buy.coinbase.com/?code=9ec56d01-7e81-5017-930c-513daa27bb6a&amount=${amount}&address=${address}&crypto_currency=ETH`
+
+    if (network === '2') {
+      url = 'https://testfaucet.metamask.io/'
+    }
+
+    chrome.tabs.create({
```

Comment:
It's ok, we have it tested this way, I'd like to deliver it working before optimizing further.

- Dan

> On Jul 21, 2016, at 12:59 PM, Frankie <notifications@github.com> wrote:
> 
> In app/scripts/metamask-controller.js:
> 
> > @@ -254,6 +266,22 @@ module.exports = class MetamaskController {
> >      this.configManager.useEtherscanProvider()
> >      extension.runtime.reload()
> >    }
> > +
> > +  buyEth (address, amount) {
> > +    if (!amount) amount = '5'
> > +
> > +    var network = this.idStore._currentState.network
> > +    var url = `https://buy.coinbase.com/?code=9ec56d01-7e81-5017-930c-513daa27bb6a&amount=${amount}&address=${address}&crypto_currency=ETH`
> > +
> > +    if (network === '2') {
> > +      url = 'https://testfaucet.metamask.io/'
> > +    }
> > +
> > +    chrome.tabs.create({
> should i move it then?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>